docs/types.rst

```diff
@@ -917,25 +917,26 @@ assigning it to a local variable, as in
 Mappings
 ========
 
-Mapping types are declared as ``mapping(_KeyType => _ValueType)``.
-Here ``_KeyType`` can be almost any type except for a mapping, a dynamically sized array, a contract, a function, an enum and a struct.
-``_ValueType`` can actually be any type, including mappings.
+You declare mapping types with the syntax ``mapping(_KeyType => _ValueType)``.
+The ``_KeyType`` can be any type except for a mapping, a dynamically sized
```

Comment:
Perhaps better to say `has to be an elementary type, i.e. you cannot use contracts, enums, structs or arrays, but dynamically-sized byte arrays and strings are fine`. (I checked that)

Reply:
Right now `strings` are fine only if the `mapping` is not `public`.